(self.webpackChunkapplied_genomics=self.webpackChunkapplied_genomics||[]).push([[8011],{3905:function(e,t,n){"use strict";n.d(t,{Zo:function(){return p},kt:function(){return w}});var r=n(7294);function o(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function a(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);t&&(r=r.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,r)}return n}function i(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?a(Object(n),!0).forEach((function(t){o(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):a(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function s(e,t){if(null==e)return{};var n,r,o=function(e,t){if(null==e)return{};var n,r,o={},a=Object.keys(e);for(r=0;r<a.length;r++)n=a[r],t.indexOf(n)>=0||(o[n]=e[n]);return o}(e,t);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);for(r=0;r<a.length;r++)n=a[r],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(o[n]=e[n])}return o}var l=r.createContext({}),c=function(e){var t=r.useContext(l),n=t;return e&&(n="function"==typeof e?e(t):i(i({},t),e)),n},p=function(e){var t=c(e.components);return r.createElement(l.Provider,{value:t},e.children)},d={inlineCode:"code",wrapper:function(e){var t=e.children;return r.createElement(r.Fragment,{},t)}},u=r.forwardRef((function(e,t){var n=e.components,o=e.mdxType,a=e.originalType,l=e.parentName,p=s(e,["components","mdxType","originalType","parentName"]),u=c(n),w=o,f=u["".concat(l,".").concat(w)]||u[w]||d[w]||a;return n?r.createElement(f,i(i({ref:t},p),{},{components:n})):r.createElement(f,i({ref:t},p))}));function w(e,t){var n=arguments,o=t&&t.mdxType;if("string"==typeof e||o){var a=n.length,i=new Array(a);i[0]=u;var s={};for(var l in t)hasOwnProperty.call(t,l)&&(s[l]=t[l]);s.originalType=e,s.mdxType="string"==typeof e?e:o,i[1]=s;for(var c=2;c<a;c++)i[c]=n[c];return r.createElement.apply(null,i)}return r.createElement.apply(null,n)}u.displayName="MDXCreateElement"},2306:function(e,t,n){"use strict";n.r(t),n.d(t,{frontMatter:function(){return i},metadata:function(){return s},toc:function(){return l},default:function(){return p}});var r=n(2122),o=n(9756),a=(n(7294),n(3905)),i={id:"mikkelsen",title:"ChIP-Seq Preprocessing",description:"Reproducing Mikkelson with modern tools",sidebar_label:"ChIP-Seq Preprocessing",sidebar_position:10},s={unversionedId:"week_06/mikkelsen",id:"week_06/mikkelsen",isDocsHomePage:!1,title:"ChIP-Seq Preprocessing",description:"Reproducing Mikkelson with modern tools",source:"@site/docs/week_06/mikkelsen.md",sourceDirName:"week_06",slug:"/week_06/mikkelsen",permalink:"/Applied-Genomics/docs/week_06/mikkelsen",editUrl:"https://github.com/Functional-Genomics-Lab/Applied-Genomics/edit/main/docs/week_06/mikkelsen.md",version:"current",sidebar_label:"ChIP-Seq Preprocessing",sidebarPosition:10,frontMatter:{id:"mikkelsen",title:"ChIP-Seq Preprocessing",description:"Reproducing Mikkelson with modern tools",sidebar_label:"ChIP-Seq Preprocessing",sidebar_position:10},sidebar:"tutorialSidebar",previous:{title:"UCSC Xena Browser",permalink:"/Applied-Genomics/docs/week_06/xena"}},l=[{value:"Finding Data",id:"finding-data",children:[{value:"Finding controls",id:"finding-controls",children:[]}]},{value:"Creating a samplesheet",id:"creating-a-samplesheet",children:[]}],c={toc:l};function p(e){var t=e.components,n=(0,o.Z)(e,["components"]);return(0,a.kt)("wrapper",(0,r.Z)({},c,n,{components:t,mdxType:"MDXLayout"}),(0,a.kt)("h2",{id:"finding-data"},"Finding Data"),(0,a.kt)("p",null,"Using SRA-explorer"),(0,a.kt)("p",null,(0,a.kt)("a",{parentName:"p",href:"https://www.ncbi.nlm.nih.gov/Traces/study/?acc=%20PRJNA106863&o=acc_s%3Aa"},"https://www.ncbi.nlm.nih.gov/Traces/study/?acc=%20PRJNA106863&o=acc_s%3Aa")),(0,a.kt)("p",null,"To download a txt file list of SRA IDs for a GEO DataSet, search\n",(0,a.kt)("a",{parentName:"p",href:"https://www.ncbi.nlm.nih.gov/geo"},"https://www.ncbi.nlm.nih.gov/geo")," using the GEO accesssion (found in the data\navailability section of published papers). Click SRA Run Selector at the bottom\nof the GEO accession page. In the SRA Run Selector sample table tick the desired\nsamples and then download the Accession List. This downloads a txt file called\nSRR_Acc_List.txt which can be used as input to ",(0,a.kt)("inlineCode",{parentName:"p"},"--public_data_ids"),"."),(0,a.kt)("h3",{id:"finding-controls"},"Finding controls"),(0,a.kt)("p",null,"Using Encode"),(0,a.kt)("h4",{id:"terms"},(0,a.kt)("a",{parentName:"h4",href:"https://www.encodeproject.org/data-standards/terms/"},"Terms")),(0,a.kt)("p",null,"Isogenic replication\n: Biological replication. Two replicates from biosamples derived from the same\nhuman donor or model organism strain. These biosamples have been treated\nseparately (i.e. two growths, two separate knockdowns, or two different\nexcisions)."),(0,a.kt)("p",null,"Technical replication\n: Two replicates from the same biosample, treated identically for each replicate\n(e.g. same growth, same knockdown)."),(0,a.kt)("h4",{id:"mef-control"},(0,a.kt)("a",{parentName:"h4",href:"https://www.encodeproject.org/experiments/ENCSR000CBB/"},"MEF Control")),(0,a.kt)("p",null,"mef_files.txt"),(0,a.kt)("pre",null,(0,a.kt)("code",{parentName:"pre",className:"language-txt"},'"https://www.encodeproject.org/metadata/?type=Experiment&accession=ENCSR000CBB&files.output_category=raw+data&files.output_type=reads&files.status=released"\nhttps://www.encodeproject.org/files/ENCFF001KPR/@@download/ENCFF001KPR.fastq.gz\nhttps://www.encodeproject.org/files/ENCFF001KPS/@@download/ENCFF001KPS.fastq.gz\n')),(0,a.kt)("pre",null,(0,a.kt)("code",{parentName:"pre",className:"language-bash"},"xargs -n 1 curl -O -L < mef_files.txt\n")),(0,a.kt)("h4",{id:"np-control"},"NP Control"),(0,a.kt)("p",null,"ENCSR295GGT_files.txt"),(0,a.kt)("pre",null,(0,a.kt)("code",{parentName:"pre",className:"language-txt"},"https://www.encodeproject.org/files/ENCFF557BAE/@@download/ENCFF557BAE.fastq.gz\nhttps://www.encodeproject.org/files/ENCFF125ETQ/@@download/ENCFF125ETQ.fastq.gz\nhttps://www.encodeproject.org/files/ENCFF066LAE/@@download/ENCFF066LAE.fastq.gz\nhttps://www.encodeproject.org/files/ENCFF078PNM/@@download/ENCFF078PNM.fastq.gz\nhttps://www.encodeproject.org/files/ENCFF202WAI/@@download/ENCFF202WAI.fastq.gz\n")),(0,a.kt)("pre",null,(0,a.kt)("code",{parentName:"pre",className:"language-bash"},"xargs -n 1 curl -O -L < ENCSR295GGT_files.txt\n")),(0,a.kt)("h4",{id:"es-control"},"ES Control"),(0,a.kt)("p",null,"es_control.txt"),(0,a.kt)("pre",null,(0,a.kt)("code",{parentName:"pre",className:"language-txt"},'"https://www.encodeproject.org/metadata/?type=Experiment&accession=ENCSR000ADJ&files.output_category=raw+data&files.output_type=reads&files.status=released"\nhttps://www.encodeproject.org/files/ENCFF001NNN/@@download/ENCFF001NNN.fastq.gz\nhttps://www.encodeproject.org/files/ENCFF001NNP/@@download/ENCFF001NNP.fastq.gz\n')),(0,a.kt)("h4",{id:"brain-control"},"Brain Control"),(0,a.kt)("p",null,"brain_control.txt"),(0,a.kt)("pre",null,(0,a.kt)("code",{parentName:"pre",className:"language-txt"},'"https://www.encodeproject.org/metadata/?type=Experiment&accession=ENCSR000CDB&files.output_category=raw+data&files.output_type=reads&files.status=released"\nhttps://www.encodeproject.org/files/ENCFF001LAO/@@download/ENCFF001LAO.fastq.gz\nhttps://www.encodeproject.org/files/ENCFF786ODW/@@download/ENCFF786ODW.fastq.gz\n')),(0,a.kt)("pre",null,(0,a.kt)("code",{parentName:"pre",className:"language-bash"},"xargs -n 1 curl -O -L < brain_control.txt\n")),(0,a.kt)("h2",{id:"creating-a-samplesheet"},"Creating a samplesheet"),(0,a.kt)("p",null,(0,a.kt)("a",{parentName:"p",href:"https://dib-lab.github.io/2020-workflows-paper/"},"https://dib-lab.github.io/2020-workflows-paper/")))}p.isMDXComponent=!0}}]);