"use strict";(self.webpackChunkapplied_genomics=self.webpackChunkapplied_genomics||[]).push([[8371],{3905:function(e,t,n){n.d(t,{Zo:function(){return p},kt:function(){return m}});var r=n(7294);function i(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function o(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);t&&(r=r.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,r)}return n}function a(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?o(Object(n),!0).forEach((function(t){i(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):o(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function l(e,t){if(null==e)return{};var n,r,i=function(e,t){if(null==e)return{};var n,r,i={},o=Object.keys(e);for(r=0;r<o.length;r++)n=o[r],t.indexOf(n)>=0||(i[n]=e[n]);return i}(e,t);if(Object.getOwnPropertySymbols){var o=Object.getOwnPropertySymbols(e);for(r=0;r<o.length;r++)n=o[r],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(i[n]=e[n])}return i}var s=r.createContext({}),c=function(e){var t=r.useContext(s),n=t;return e&&(n="function"==typeof e?e(t):a(a({},t),e)),n},p=function(e){var t=c(e.components);return r.createElement(s.Provider,{value:t},e.children)},u={inlineCode:"code",wrapper:function(e){var t=e.children;return r.createElement(r.Fragment,{},t)}},d=r.forwardRef((function(e,t){var n=e.components,i=e.mdxType,o=e.originalType,s=e.parentName,p=l(e,["components","mdxType","originalType","parentName"]),d=c(n),m=i,b=d["".concat(s,".").concat(m)]||d[m]||u[m]||o;return n?r.createElement(b,a(a({ref:t},p),{},{components:n})):r.createElement(b,a({ref:t},p))}));function m(e,t){var n=arguments,i=t&&t.mdxType;if("string"==typeof e||i){var o=n.length,a=new Array(o);a[0]=d;var l={};for(var s in t)hasOwnProperty.call(t,s)&&(l[s]=t[s]);l.originalType=e,l.mdxType="string"==typeof e?e:i,a[1]=l;for(var c=2;c<o;c++)a[c]=n[c];return r.createElement.apply(null,a)}return r.createElement.apply(null,n)}d.displayName="MDXCreateElement"},1018:function(e,t,n){n.r(t),n.d(t,{assets:function(){return s},contentTitle:function(){return a},default:function(){return u},frontMatter:function(){return o},metadata:function(){return l},toc:function(){return c}});var r=n(3117),i=(n(7294),n(3905));const o={id:"scrnaseq",title:"scRNA-Seq",description:"",sidebar_label:"scRNA-Seq",sidebar_position:2},a=void 0,l={unversionedId:"misc/scrnaseq",id:"version-22u/misc/scrnaseq",title:"scRNA-Seq",description:"",source:"@site/versioned_docs/version-22u/misc/scRNA-seq.md",sourceDirName:"misc",slug:"/misc/scrnaseq",permalink:"/Applied-Genomics/docs/22u/misc/scrnaseq",draft:!1,editUrl:"https://github.com/Functional-Genomics-Lab/Applied-Genomics/edit/main/versioned_docs/version-22u/misc/scRNA-seq.md",tags:[],version:"22u",sidebarPosition:2,frontMatter:{id:"scrnaseq",title:"scRNA-Seq",description:"",sidebar_label:"scRNA-Seq",sidebar_position:2},sidebar:"tutorialSidebar",previous:{title:"Conda",permalink:"/Applied-Genomics/docs/22u/misc/conda"},next:{title:"Xena Browser",permalink:"/Applied-Genomics/docs/22u/misc/xena"}},s={},c=[{value:"GitHub Repo",id:"github-repo",level:4},{value:"Paper",id:"paper",level:4},{value:"Adjustments to run it on sysbio",id:"adjustments-to-run-it-on-sysbio",level:2}],p={toc:c};function u(e){let{components:t,...n}=e;return(0,i.kt)("wrapper",(0,r.Z)({},p,n,{components:t,mdxType:"MDXLayout"}),(0,i.kt)("h4",{id:"github-repo"},"GitHub Repo"),(0,i.kt)("p",null,(0,i.kt)("a",{parentName:"p",href:"https://github.com/theislab/single-cell-tutorial"},"theislab/single-cell-tutorial")),(0,i.kt)("h4",{id:"paper"},"Paper"),(0,i.kt)("p",null,(0,i.kt)("a",{parentName:"p",href:"https://pubmed.ncbi.nlm.nih.gov/31217225/"},"Current best practices in single-cell RNA-seq analysis: a tutorial")),(0,i.kt)("h2",{id:"adjustments-to-run-it-on-sysbio"},"Adjustments to run it on sysbio"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-bash"},"login $ git clone https://github.com/theislab/single-cell-tutorial.git\nlogin $ cd single-cell-tutorial\n")),(0,i.kt)("p",null,"Download the example data (This is all on the login node)"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-bash"},"mkdir -p data/Haber-et-al_mouse-intestinal-epithelium/\ncd data/Haber-et-al_mouse-intestinal-epithelium/\nwget ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92332/suppl/GSE92332_RAW.tar\nmkdir GSE92332_RAW\ntar -C GSE92332_RAW -xvf GSE92332_RAW.tar\ngunzip GSE92332_RAW/*_Regional_*\nwget ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE92nnn/GSE92332/suppl/GSE92332_Regional_UMIcounts.txt.gz\ngunzip GSE92332_Regional_UMIcounts.txt.gz\n")),(0,i.kt)("p",null,"Start the container on a compute node"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-bash"},"login $ srun --pty bash\nsysbio-0 $ ml load singularity\nsysbio-0 $ singularity exec /scratch/applied-genomics/scrnaseq/single-cell-analysis_210114.sif bash\n")),(0,i.kt)("p",null,"and then paste the following in to set the environment properly:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-bash"},"export LD_LIBRARY_PATH=/opt/R/lib/R/lib\nexport SHELL=/bin/bash\nexport TERM=xterm-256color\nexport LANG=en_US.UTF-8\necho $PATH\nexport PATH=/opt/python/bin:/opt/R/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin\n")),(0,i.kt)("p",null,"and fire up jupyterlab"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-bash"},"jupyter-lab --no-browser --ip=0.0.0.0\n")),(0,i.kt)("ol",{start:4},(0,i.kt)("li",{parentName:"ol"},"Open another terminal and login to the compute node with port forwarding.")),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-bash"},"login $ ssh sysbio-1 -L 8888:localhost:8888\n")),(0,i.kt)("ol",{start:5},(0,i.kt)("li",{parentName:"ol"},"Go back to the first terminal and copy the jupyterlab link into your local browser. It should look something like ",(0,i.kt)("inlineCode",{parentName:"li"},"http://localhost:8888/lab?token=4d08d209468541b333109fc4f957c8b3d1c97bed59072148"))),(0,i.kt)("p",null,"When you finish consider cleaning out the example data to save space on the server"))}u.isMDXComponent=!0}}]);