<!doctype html>
<html lang="en" dir="ltr">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.0">
<link rel="alternate" type="application/rss+xml" href="/Applied-Genomics/blog/rss.xml" title="Applied Genomics Blog RSS Feed">
<link rel="alternate" type="application/atom+xml" href="/Applied-Genomics/blog/atom.xml" title="Applied Genomics Blog Atom Feed"><title data-react-helmet="true">Assignment 3 | Applied Genomics</title><meta data-react-helmet="true" property="og:url" content="https://your-docusaurus-test-site.com/Applied-Genomics/docs/week_03/assignment_03"><meta data-react-helmet="true" name="docusaurus_locale" content="en"><meta data-react-helmet="true" name="docusaurus_version" content="21u"><meta data-react-helmet="true" name="docusaurus_tag" content="docs-default-21u"><meta data-react-helmet="true" property="og:title" content="Assignment 3 | Applied Genomics"><meta data-react-helmet="true" name="description" content="Summary: Create a counts file for SARSâ€‘CoVâ€‘2 alignment and genes."><meta data-react-helmet="true" property="og:description" content="Summary: Create a counts file for SARSâ€‘CoVâ€‘2 alignment and genes."><link data-react-helmet="true" rel="shortcut icon" href="/Applied-Genomics/img/favicon.ico"><link data-react-helmet="true" rel="canonical" href="https://your-docusaurus-test-site.com/Applied-Genomics/docs/week_03/assignment_03"><link data-react-helmet="true" rel="alternate" href="https://your-docusaurus-test-site.com/Applied-Genomics/docs/week_03/assignment_03" hreflang="en"><link data-react-helmet="true" rel="alternate" href="https://your-docusaurus-test-site.com/Applied-Genomics/docs/week_03/assignment_03" hreflang="x-default"><link rel="stylesheet" href="/Applied-Genomics/assets/css/styles.97b09745.css">
<link rel="preload" href="/Applied-Genomics/assets/js/runtime~main.59a94dff.js" as="script">
<link rel="preload" href="/Applied-Genomics/assets/js/main.a9510e8b.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div><a href="#main" class="skipToContent_1oUP">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle" type="button" tabindex="0"><svg aria-label="Menu" width="30" height="30" viewBox="0 0 30 30" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/Applied-Genomics/"><strong class="navbar__title">Applied Genomics</strong></a><a class="navbar__item navbar__link navbar__link--active" href="/Applied-Genomics/docs/overview">Course</a><a class="navbar__item navbar__link" href="/Applied-Genomics/docs/resources">Resources</a></div><div class="navbar__items navbar__items--right"><a href="http://taehoonkim.org/" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">Functional Genomics Laboratory</a><div class="react-toggle displayOnlyInLargeViewport_GrZ2 react-toggle--disabled" role="button" tabindex="-1"><div class="react-toggle-track"><div class="react-toggle-track-check"><span class="toggle_71bT">ðŸŒœ</span></div><div class="react-toggle-track-x"><span class="toggle_71bT">ðŸŒž</span></div></div><div class="react-toggle-thumb"></div><input type="checkbox" class="react-toggle-screenreader-only" aria-label="Switch between dark and light mode"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div><div class="navbar-sidebar"><div class="navbar-sidebar__brand"><a class="navbar__brand" href="/Applied-Genomics/"><strong class="navbar__title">Applied Genomics</strong></a></div><div class="navbar-sidebar__items"><div class="menu"><ul class="menu__list"><li class="menu__list-item"><a class="menu__link navbar__link--active" href="/Applied-Genomics/docs/overview">Course</a></li><li class="menu__list-item"><a class="menu__link" href="/Applied-Genomics/docs/resources">Resources</a></li><li class="menu__list-item"><a href="http://taehoonkim.org/" target="_blank" rel="noopener noreferrer" class="menu__link">Functional Genomics Laboratory</a></li></ul></div></div></div></nav><div class="main-wrapper docs-wrapper doc-page"><div class="docPage_31aa"><div class="docSidebarContainer_3Kbt" role="complementary"><div class="sidebar_15mo"><div class="menu menu--responsive thin-scrollbar menu_Bmed"><button aria-label="Open menu" aria-haspopup="true" class="button button--secondary button--sm menu__button" type="button"><svg aria-label="Menu" class="sidebarMenuIcon_fgN0" width="24" height="24" viewBox="0 0 30 30" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" href="/Applied-Genomics/docs/resources">Resources</a></li><li class="menu__list-item"><a class="menu__link" href="/Applied-Genomics/docs/overview">Overview</a></li><li class="menu__list-item"><a class="menu__link" href="/Applied-Genomics/docs/week-1">Week 1</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Week 2</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/week_02/intro">Reproducible Research</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/week_02/git">Version Control Systems</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/week_02/make">Make</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/week_02/documentation">Documentation</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/week_02/jupyter">Jupyter</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/week_02/assignment_02">Assignment 2</a></li></ul></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#!">Week 3</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/Applied-Genomics/docs/week_03/code_alternatives">VS Code Alternatives</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/Applied-Genomics/docs/week_03/conda">Conda</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/Applied-Genomics/docs/week_03/hpc">HPC</a></li><li class="menu__list-item"><a aria-current="page" class="menu__link menu__link--active active" tabindex="0" href="/Applied-Genomics/docs/week_03/assignment_03">Assignment 3</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Week 4</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/week_04/transcriptomics">Transcriptomics</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/week_04/nf-core">nf-core</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Week 5</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/week_05/jupyter_slurm">Jupyter on Slurm</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/week_05/scrnaseq">scRNA-Seq</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/week_05/assignment_05">Assignment 5</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Week 6</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/week_06/ucsc">UCSC Genome Browser</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/week_06/xena">Xena Browser</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/week_06/mikkelsen">ChIP-Seq Preprocessing</a></li></ul></li></ul></div></div></div><main class="docMainContainer_3ufF"><div class="container padding-vert--lg docItemWrapper_3FMP"><div class="row"><div class="col docItemCol_3FnS"><div class="docItemContainer_33ec"><article><div><span class="badge badge--secondary">Version: 21u</span></div><header><h1 class="docTitle_3a4h">Assignment 3</h1></header><div class="markdown"><p>Summary: Create a counts file for SARSâ€‘CoVâ€‘2 alignment and genes.</p><p>Things that will be checked for:</p><ol><li>Make rules to download the data (<strong>2 points</strong>)
<a href="https://github.com/nf-core/test-datasets/raw/modules/data/genomics/sarscov2/genome/genome.gtf" target="_blank" rel="noopener noreferrer">https://github.com/nf-core/test-datasets/raw/modules/data/genomics/sarscov2/genome/genome.gtf</a><a href="https://github.com/nf-core/test-datasets/raw/modules/data/genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam" target="_blank" rel="noopener noreferrer">https://github.com/nf-core/test-datasets/raw/modules/data/genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam</a></li></ol><p>You&#x27;ll need to use the -O flag to change the name of the output file with wget. Here&#x27;s one of the rules as an example</p><div class="codeBlockContainer_K1bP"><div class="codeBlockContent_hGly Makefile"><div tabindex="0" class="prism-code language-Makefile codeBlock_23N8 thin-scrollbar"><div class="codeBlockLines_39YC" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">sars_genome.gtf:</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">    wget -O cov_genome.gtf https://github.com/nf-core/test-datasets/raw/modules/data/genomics/sarscov2/genome/genome.gtf</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_Ue-o">Copy</button></div></div><ol start="2"><li>Add a second batch script (<code>submit_cov.sh</code> works) to run featurecounts with the sarscov2 data (<strong>2 points</strong>)</li></ol><p>You&#x27;ll need to use <code>featureCounts -t gene ...</code> instead of <code>featureCounts -t exon ...</code> because there aren&#x27;t any exons in the cov_genome.gtf</p><ol start="3"><li>Create 3 new make rules, one to run the &quot;test&quot; batch job, one to run the &quot;sarscov2&quot; data, and at the top an all rule, that should call the download targets and then the job targets like below (<strong>4 points</strong>)</li></ol><div class="codeBlockContainer_K1bP"><div class="codeBlockContent_hGly Makefile"><div tabindex="0" class="prism-code language-Makefile codeBlock_23N8 thin-scrollbar"><div class="codeBlockLines_39YC" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">all: download_data featurecounts</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_Ue-o">Copy</button></div></div><ol start="4"><li>Submit your cov_counts.txt file on eLearning (<strong>2 points</strong>)</li></ol><p>Choose one only one bonus question(if any), you won&#x27;t get extra credit for doing both.</p><p>Bonus 1 (<strong>5 points</strong>):</p><p>Generalize your make rules for feature counts with wildcards</p><ul><li><a href="http://web.mit.edu/gnu/doc/html/make_4.html#SEC22" target="_blank" rel="noopener noreferrer">http://web.mit.edu/gnu/doc/html/make_4.html#SEC22</a></li></ul><p>Bonus 2 (<strong>5 points</strong>):</p><p>Use slurm job arrays to use one only one batch script for both of the featureCount runs. Some resources(note this wasn&#x27;t covered in class):</p><ul><li><a href="https://github.com/Functional-Genomics-Lab/useful-scripts/blob/e377885121dc62116c07d7f2cb32fbaac7935be4/bwa.sh#L3" target="_blank" rel="noopener noreferrer">https://github.com/Functional-Genomics-Lab/useful-scripts/blob/e377885121dc62116c07d7f2cb32fbaac7935be4/bwa.sh#L3</a></li><li><a href="https://www.biostars.org/p/338078/" target="_blank" rel="noopener noreferrer">https://www.biostars.org/p/338078/</a></li><li><a href="https://slurm.schedmd.com/job_array.html" target="_blank" rel="noopener noreferrer">https://slurm.schedmd.com/job_array.html</a></li></ul><p>You can use -t gene with the homo_sapians data set we used in class as well.</p></div></article><div class="margin-vert--xl"><div class="row"><div class="col"><a href="https://github.com/Functional-Genomics-Lab/Applied-Genomics/edit/main/versioned_docs/version-21u/week_03/assignment_03.md" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="1.2em" width="1.2em" preserveAspectRatio="xMidYMid meet" role="img" viewBox="0 0 40 40" class="iconEdit_2_ui" aria-label="Edit page"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div></div></div><div class="margin-vert--lg"><nav class="pagination-nav" aria-label="Docs pages navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/Applied-Genomics/docs/week_03/hpc"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Â« High Performance Computing</div></a></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/Applied-Genomics/docs/week_04/transcriptomics"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Transcriptomics Â»</div></a></div></nav></div></div></div><div class="col col--3"><div class="tableOfContents_35-E thin-scrollbar"></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="row footer__links"><div class="col footer__col"><h4 class="footer__title">Course</h4><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/Applied-Genomics/docs/overview">Overview</a></li></ul></div></div><div class="footer__bottom text--center"></div></div></footer></div>
<script src="/Applied-Genomics/assets/js/runtime~main.59a94dff.js"></script>
<script src="/Applied-Genomics/assets/js/main.a9510e8b.js"></script>
</body>
</html>