<!doctype html>
<html lang="en" dir="ltr">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.0">
<link rel="alternate" type="application/rss+xml" href="/Applied-Genomics/blog/rss.xml" title="Applied Genomics Blog RSS Feed">
<link rel="alternate" type="application/atom+xml" href="/Applied-Genomics/blog/atom.xml" title="Applied Genomics Blog Atom Feed"><title data-react-helmet="true">High Performance Computing | Applied Genomics</title><meta data-react-helmet="true" property="og:url" content="https://your-docusaurus-test-site.com/Applied-Genomics/docs/21u/week_03/hpc"><meta data-react-helmet="true" name="docusaurus_locale" content="en"><meta data-react-helmet="true" name="docusaurus_version" content="21u"><meta data-react-helmet="true" name="docusaurus_tag" content="docs-default-21u"><meta data-react-helmet="true" property="og:title" content="High Performance Computing | Applied Genomics"><meta data-react-helmet="true" name="description" content="How to use big computers"><meta data-react-helmet="true" property="og:description" content="How to use big computers"><link data-react-helmet="true" rel="shortcut icon" href="/Applied-Genomics/img/favicon.ico"><link data-react-helmet="true" rel="canonical" href="https://your-docusaurus-test-site.com/Applied-Genomics/docs/21u/week_03/hpc"><link data-react-helmet="true" rel="alternate" href="https://your-docusaurus-test-site.com/Applied-Genomics/docs/21u/week_03/hpc" hreflang="en"><link data-react-helmet="true" rel="alternate" href="https://your-docusaurus-test-site.com/Applied-Genomics/docs/21u/week_03/hpc" hreflang="x-default"><link rel="stylesheet" href="/Applied-Genomics/assets/css/styles.97b09745.css">
<link rel="preload" href="/Applied-Genomics/assets/js/runtime~main.70778c13.js" as="script">
<link rel="preload" href="/Applied-Genomics/assets/js/main.c419694f.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div><a href="#main" class="skipToContent_1oUP">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle" type="button" tabindex="0"><svg aria-label="Menu" width="30" height="30" viewBox="0 0 30 30" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/Applied-Genomics/"><strong class="navbar__title">Applied Genomics</strong></a><a class="navbar__item navbar__link navbar__link--active" href="/Applied-Genomics/docs/21u/overview">Course</a><a class="navbar__item navbar__link" href="/Applied-Genomics/docs/resources">Resources</a></div><div class="navbar__items navbar__items--right"><div class="navbar__item dropdown dropdown--hoverable dropdown--right"><a class="navbar__item navbar__link" href="/Applied-Genomics/docs/21u/resources">Summer 21</a><ul class="dropdown__menu"><li><a class="dropdown__link" href="/Applied-Genomics/docs/week_03/hpc">Summer 22</a></li><li><a aria-current="page" class="dropdown__link dropdown__link--active" href="/Applied-Genomics/docs/21u/week_03/hpc">Summer 21</a></li><li><a class="dropdown__link" href="/Applied-Genomics/versions">All versions</a></li></ul></div><a href="http://taehoonkim.org/" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">Functional Genomics Laboratory</a><div class="react-toggle displayOnlyInLargeViewport_GrZ2 react-toggle--disabled" role="button" tabindex="-1"><div class="react-toggle-track"><div class="react-toggle-track-check"><span class="toggle_71bT">🌜</span></div><div class="react-toggle-track-x"><span class="toggle_71bT">🌞</span></div></div><div class="react-toggle-thumb"></div><input type="checkbox" class="react-toggle-screenreader-only" aria-label="Switch between dark and light mode"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div><div class="navbar-sidebar"><div class="navbar-sidebar__brand"><a class="navbar__brand" href="/Applied-Genomics/"><strong class="navbar__title">Applied Genomics</strong></a></div><div class="navbar-sidebar__items"><div class="menu"><ul class="menu__list"><li class="menu__list-item"><a class="menu__link navbar__link--active" href="/Applied-Genomics/docs/21u/overview">Course</a></li><li class="menu__list-item"><a class="menu__link" href="/Applied-Genomics/docs/resources">Resources</a></li><li class="menu__list-item"><a role="button" class="menu__link menu__link--sublist">Versions</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" href="/Applied-Genomics/docs/week_03/hpc">Summer 22</a></li><li class="menu__list-item"><a aria-current="page" class="menu__link menu__link--active" href="/Applied-Genomics/docs/21u/week_03/hpc">Summer 21</a></li><li class="menu__list-item"><a class="menu__link" href="/Applied-Genomics/versions">All versions</a></li></ul></li><li class="menu__list-item"><a href="http://taehoonkim.org/" target="_blank" rel="noopener noreferrer" class="menu__link">Functional Genomics Laboratory</a></li></ul></div></div></div></nav><div class="main-wrapper docs-wrapper doc-page"><div class="docPage_31aa"><div class="docSidebarContainer_3Kbt" role="complementary"><div class="sidebar_15mo"><div class="menu menu--responsive thin-scrollbar menu_Bmed"><button aria-label="Open menu" aria-haspopup="true" class="button button--secondary button--sm menu__button" type="button"><svg aria-label="Menu" class="sidebarMenuIcon_fgN0" width="24" height="24" viewBox="0 0 30 30" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" href="/Applied-Genomics/docs/21u/resources">Resources</a></li><li class="menu__list-item"><a class="menu__link" href="/Applied-Genomics/docs/21u/overview">Overview</a></li><li class="menu__list-item"><a class="menu__link" href="/Applied-Genomics/docs/21u/week-1">Week 1</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Week 2</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/21u/week_02/intro">Reproducible Research</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/21u/week_02/git">Version Control Systems</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/21u/week_02/make">Make</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/21u/week_02/documentation">Documentation</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/21u/week_02/jupyter">Jupyter</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/21u/week_02/assignment_02">Assignment 2</a></li></ul></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#!">Week 3</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/Applied-Genomics/docs/21u/week_03/code_alternatives">VS Code Alternatives</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/Applied-Genomics/docs/21u/week_03/conda">Conda</a></li><li class="menu__list-item"><a aria-current="page" class="menu__link menu__link--active active" tabindex="0" href="/Applied-Genomics/docs/21u/week_03/hpc">HPC</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/Applied-Genomics/docs/21u/week_03/assignment_03">Assignment 3</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Week 4</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/21u/week_04/transcriptomics">Transcriptomics</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/21u/week_04/nf-core">nf-core</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Week 5</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/21u/week_05/jupyter_slurm">Jupyter on Slurm</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/21u/week_05/scrnaseq">scRNA-Seq</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/21u/week_05/assignment_05">Assignment 5</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Week 6</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/21u/week_06/ucsc">UCSC Genome Browser</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/21u/week_06/xena">Xena Browser</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/Applied-Genomics/docs/21u/week_06/mikkelsen">ChIP-Seq Preprocessing</a></li></ul></li></ul></div></div></div><main class="docMainContainer_3ufF"><div class="container padding-vert--lg docItemWrapper_3FMP"><div class="row"><div class="col docItemCol_3FnS"><div class="alert alert--warning margin-bottom--md" role="alert"><div>This is documentation for Applied Genomics <strong>Summer 21</strong>, which is no longer actively maintained.</div><div class="margin-top--md">For up-to-date documentation, see the <strong><a href="/Applied-Genomics/docs/week_03/hpc">latest version</a></strong> (Summer 22).</div></div><div class="docItemContainer_33ec"><article><div><span class="badge badge--secondary">Version: Summer 21</span></div><header><h1 class="docTitle_3a4h">High Performance Computing</h1></header><div class="markdown"><p>In the era of big data, we need big computers. Just like other expensive
resources these are shared among many researchers, however this also makes them
easy to get access to. At UT Dallas alone we have a few clusters you can
utilize:</p><ul><li>Sysbio, for training</li><li>Ganymede, for <strong>H</strong>igh <strong>P</strong>erformance <strong>C</strong>omputing</li><li>Europa(Public access coming soon), for <strong>H</strong>igh <strong>T</strong>hroughput <strong>C</strong>omputing</li></ul><p>Just for comparison, a node in the genomics queue on Ganymede has 16 cores, 32GB
of RAM, and is connected to a petabyte of storage. There&#x27;s also 16 nodes.
They&#x27;re free for you to use as well! You just need to know how to use them.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="whats-a-cluster"></a>What&#x27;s a cluster?<a class="hash-link" href="#whats-a-cluster" title="Direct link to heading">#</a></h2><p><img src="https://portal.tacc.utexas.edu/documents/10157/1181317/Login+and+compute+nodes/dd6fa98c-1695-4e62-8b7b-66f0c83ceba3?t=1436213020000" alt="TACC Graphic"><sup id="fnref-1"><a href="#fn-1" class="footnote-ref">1</a></sup></p><ul><li>You login to the login node<ul><li>Login nodes are a prep area</li><li>Transferring files</li><li>Compiling code</li><li>Submitting jobs</li></ul></li><li>The compute nodes are where actual computations occur and where research is
done. All batch jobs and executables, as well as development and debugging
sessions, are run on the compute nodes.<sup id="fnref-1"><a href="#fn-1" class="footnote-ref">1</a></sup></li></ul><p><img src="https://docs.ycrc.yale.edu/img/clusters.png" alt="Yale Shared File System"><sup id="fnref-2"><a href="#fn-2" class="footnote-ref">2</a></sup></p><ul><li>The nodes all have access to a shared filesystem</li></ul><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="slurm"></a>Slurm<a class="hash-link" href="#slurm" title="Direct link to heading">#</a></h2><p>SLURM - Simple Linux Utility for Resource Management</p><p>Provides a way for users to schedule jobs. There are plenty of alternatives but
Slurm is pretty common. Checkout this <a href="https://bihealth.github.io/bih-cluster/slurm/cheat-sheet/" target="_blank" rel="noopener noreferrer">Slurm
Cheatsheet</a> if you
need to reference the commands to control it.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="tutorial"></a>Tutorial<a class="hash-link" href="#tutorial" title="Direct link to heading">#</a></h2><div class="admonition admonition-info alert alert--info"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="14" height="16" viewBox="0 0 14 16"><path fill-rule="evenodd" d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg></span>info</h5></div><div class="admonition-content"><p>Before we start with our first steps tutorial, we would like to introduce the following convention that we use throughout the series:</p><div class="codeBlockContainer_K1bP"><div class="codeBlockContent_hGly bash"><div tabindex="0" class="prism-code language-bash codeBlock_23N8 thin-scrollbar"><div class="codeBlockLines_39YC" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">$ Commands are prefixed with a little dollar sign</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">login $ The </span><span class="token function" style="color:rgb(130, 170, 255)">host</span><span class="token plain"> name is put infront of the dollar sign</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">compute $</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_Ue-o">Copy</button></div></div></div></div><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="instant-gratification"></a>Instant Gratification<a class="hash-link" href="#instant-gratification" title="Direct link to heading">#</a></h3><div class="codeBlockContainer_K1bP"><div class="codeBlockContent_hGly bash"><div tabindex="0" class="prism-code language-bash codeBlock_23N8 thin-scrollbar"><div class="codeBlockLines_39YC" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">login $ srun -p normal --time </span><span class="token number" style="color:rgb(247, 140, 108)">1</span><span class="token plain">-00 --mem</span><span class="token operator" style="color:rgb(137, 221, 255)">=</span><span class="token plain">8G --ntasks</span><span class="token operator" style="color:rgb(137, 221, 255)">=</span><span class="token number" style="color:rgb(247, 140, 108)">8</span><span class="token plain"> --pty </span><span class="token function" style="color:rgb(130, 170, 255)">bash</span><span class="token plain"> -i</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic"># Once you&#x27;re logged into a node</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">compute $ </span><span class="token builtin class-name" style="color:rgb(255, 203, 107)">exit</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_Ue-o">Copy</button></div></div><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="scratch"></a><code>scratch</code><a class="hash-link" href="#scratch" title="Direct link to heading">#</a></h3><p>All users on the system have a file system quota. You&#x27;ll see it every time you
login. There&#x27;s a <strong>soft</strong> limit, that if you go over you&#x27;ll get the warning: <code> dm-0: warning, user block quota exceeded.</code>. There&#x27;s also a Hard limit, which if
you exceed it the process will be killed. You can check your quota with <code>quota -s</code></p><p><a href="http://docs.oithpc.utdallas.edu/#ganymede-space-constraints" target="_blank" rel="noopener noreferrer">See Ganymede Space Constraints for more info</a></p><p>But you should notice in your home directory you have a symbolic link called
<code>scratch</code>. This is connected to a large storage system and is unmetered.</p><p>Here&#x27;s a quick table comparing the two:</p><table><thead><tr><th>Attributes</th><th>Home Space</th><th>Scratch Space</th></tr></thead><tbody><tr><td>Purpose of the Space</td><td>To house scripts, source code and output data.</td><td>To have a large area for working with data or trying ideas</td></tr><tr><td>Backed Up</td><td>Yes</td><td>No</td></tr></tbody></table><p><sup id="fnref-3"><a href="#fn-3" class="footnote-ref">3</a></sup></p><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="lets-get-some-data"></a>Let&#x27;s get some data<a class="hash-link" href="#lets-get-some-data" title="Direct link to heading">#</a></h3><p>So how do we get data from our browser, to the computer? How do we avoid
downloading it locally(to our laptops, which have limited space)?</p><h4><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="wget-the-non-interactive-network-downloader"></a>wget-The non-interactive network downloader<a class="hash-link" href="#wget-the-non-interactive-network-downloader" title="Direct link to heading">#</a></h4><ul><li>wget allows us to download files from the web via the command line</li><li>It supports http, https and ftp</li></ul><p>Let&#x27;s use it!</p><ol><li>Let&#x27;s clone our project into the scratch directory</li></ol><div class="codeBlockContainer_K1bP"><div class="codeBlockContent_hGly bash"><div tabindex="0" class="prism-code language-bash codeBlock_23N8 thin-scrollbar"><div class="codeBlockLines_39YC" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">login $ </span><span class="token builtin class-name" style="color:rgb(255, 203, 107)">cd</span><span class="token plain"> scratch</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">login $ </span><span class="token function" style="color:rgb(130, 170, 255)">git</span><span class="token plain"> clone https://github.com/github-username/ag-intro.git</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">login $ </span><span class="token builtin class-name" style="color:rgb(255, 203, 107)">cd</span><span class="token plain"> ag-intro</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_Ue-o">Copy</button></div></div><ol start="2"><li>Download the data.</li></ol><div class="codeBlockContainer_K1bP"><div class="codeBlockContent_hGly bash"><div tabindex="0" class="prism-code language-bash codeBlock_23N8 thin-scrollbar"><div class="codeBlockLines_39YC" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">login $ </span><span class="token function" style="color:rgb(130, 170, 255)">wget</span><span class="token plain"> https://github.com/nf-core/test-datasets/raw/modules/data/genomics/homo_sapiens/genome/genome.gtf</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">login $ </span><span class="token function" style="color:rgb(130, 170, 255)">wget</span><span class="token plain"> https://github.com/nf-core/test-datasets/raw/modules/data/genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_Ue-o">Copy</button></div></div><p>While we&#x27;re at it, let&#x27;s through these commands into our <code>Makefile</code> so we don&#x27;t forget the commands!</p><div class="codeBlockContainer_K1bP"><div class="codeBlockContent_hGly Makefile"><div tabindex="0" class="prism-code language-Makefile codeBlock_23N8 thin-scrollbar"><div class="codeBlockLines_39YC" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">download_data: genome.gtf test.paired_end.sorted.bam</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">genome.gtf:</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">    wget https://github.com/nf-core/test-datasets/raw/modules/data/genomics/homo_sapiens/genome/genome.gtf</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">test.paired_end.sorted.bam:</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">    wget https://github.com/nf-core/test-datasets/raw/modules/data/genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_Ue-o">Copy</button></div></div><p>Don&#x27;t forget to add the data to <code>.gitignore</code> before you commit!</p><h5><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="what-did-we-just-download"></a>What did we just download?<a class="hash-link" href="#what-did-we-just-download" title="Direct link to heading">#</a></h5><p>These files are two different formats that are common:</p><blockquote><p>The GFF/GTF/BED formats are the so-called “interval” formats that retain only the coordinate positions for a region in a genome. Each is tab delimited and will contain information on the chromosomal coordinate, start, end, strand, value and other attributes, though the order of the columns will depend on the feature.<sup id="fnref-4"><a href="#fn-4" class="footnote-ref">4</a></sup></p></blockquote><p>The GFF/GTF formats are 9 column tab-delimited formats. The coordinates start at value 1.</p><div class="codeBlockContainer_K1bP"><div class="codeBlockContent_hGly tsv"><div tabindex="0" class="prism-code language-tsv codeBlock_23N8 thin-scrollbar"><div class="codeBlockLines_39YC" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">chr1 . mRNA            1300  9000  .  +  .  ID=mrna0001;Name=sonichedgehog</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">chr1 . exon            1300  1500  .  +  .  ID=exon00001;Parent=mrna0001</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">chr1 . exon            1050  1500  .  +  .  ID=exon00002;Parent=mrna0001</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_Ue-o">Copy</button></div></div><blockquote><p>The SAM/BAM formats are so-called Sequence Alignment Maps. These files typically represent the results of aligning a FASTQ file to a reference FASTA file and describe the individual, pairwise alignments that were found. Different algorithms may create different alignments (and hence BAM files).<sup id="fnref-4"><a href="#fn-4" class="footnote-ref">4</a></sup></p></blockquote><div class="codeBlockContainer_K1bP"><div class="codeBlockContent_hGly tsv"><div tabindex="0" class="prism-code language-tsv codeBlock_23N8 thin-scrollbar"><div class="codeBlockLines_39YC" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">SRR1553425.13617    163 AF086833.2  46  60  101M    =   541 596 GAATAACTATGAGGAAGATTAATAATTTTCCTCTCATTGAAATTTATATCGGAATTTAAATTGAAATTGTTACTGTAATCATACCTGGTTTGTTTCAGAGC   @&lt;@FFFDDHHGHHJIJJIJJJJJJJJJJJJJJIIJJJJGIJJJJJJEIEIGIIJJJJJIIIIGIIIGGEGIIIGIIJJJGGGHHFHGFDDFDEEDDEDCDCNM:i:1 MD:Z:81C19  MC:Z:101M   AS:i:96 XS:i:0</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">SRR1553425.13755    99  AF086833.2  46  60  101M    =   46  101 GAATAACTATGAGGAAGATTAATAATTTTCCTCTCATTGAAATTTATATCGGAATTTAAATTGAAATTGTTACTGTAATCATACCTGGTTTGTTTCAGAGC   ?:@DBDDDFHGDHGIGEGAFGGHDHIIIIIIIGGGGIGEBHHGHFGHIDCHIIIIII&lt;G@FGHDFGGHH@FGGEGGGEHIE@HHHGHF;?BC@CECFC:@&gt;NM:i:1 MD:Z:81C19  MC:Z:101M   AS:i:96 XS:i:0</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">SRR1553425.13755    147 AF086833.2  46  60  101M    =   46  -101    GAATAACTATGAGGAAGATTAATAATTTTCCTCTCATTGAAATTTATATCGGAATTTAAATTGAAATTGTTACTGTAATCATACCTGGTTTGTTTCAGAGC   DDDDCDDDDDECEEDEDFFFEFFGHHHHIGGHG@=@@=GIIJHIIHFDIIIIIIHHGJIFAJJJIGIJIIIJHHGGIGEGGHFHHJIIHHHGHFFFFFCCCNM:i:1 MD:Z:81C19  MC:Z:101M   AS:i:96 XS:i:0</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">SRR1553425.11219    2227    AF086833.2  47  60  71H30M  =   146 171 AATAACTATGAGGAAGATTAATAATTTTCC  IGHF@GHEHIIHE&gt;IIGHFDHDDDFFF@@@  NM:i:0  MD:Z:30 MC:Z:101M   AS:i:30 XS:i:0  SA:Z:AF086833.2,68,+,30S71M,60,1;</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_Ue-o">Copy</button></div></div><p>You can find more info at <a href="https://www.biostarhandbook.com/common-data-types.html" target="_blank" rel="noopener noreferrer">Biostars handbook Common data types</a></p><h5><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="some-other-ways-to-get-data-onto-the-cluster"></a>Some other ways to get data onto the cluster<a class="hash-link" href="#some-other-ways-to-get-data-onto-the-cluster" title="Direct link to heading">#</a></h5><p><code>wget</code> isn&#x27;t the only way to get things to the cluster. It&#x27;s great for stuff that&#x27;s on the internet, but what about local files?<sup id="fnref-5"><a href="#fn-5" class="footnote-ref">5</a></sup></p><h6><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="scp---secure-copy"></a>SCP - secure copy<a class="hash-link" href="#scp---secure-copy" title="Direct link to heading">#</a></h6><ul><li>Local file to remote server</li></ul><div class="codeBlockContainer_K1bP"><div class="codeBlockContent_hGly bash"><div tabindex="0" class="prism-code language-bash codeBlock_23N8 thin-scrollbar"><div class="codeBlockLines_39YC" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token function" style="color:rgb(130, 170, 255)">scp</span><span class="token plain"> filename.txt hostname:/path/to/filename/filename.txt</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_Ue-o">Copy</button></div></div><ul><li>Remote server to local file</li></ul><div class="codeBlockContainer_K1bP"><div class="codeBlockContent_hGly bash"><div tabindex="0" class="prism-code language-bash codeBlock_23N8 thin-scrollbar"><div class="codeBlockLines_39YC" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token function" style="color:rgb(130, 170, 255)">scp</span><span class="token plain"> hostname:filename.txt </span><span class="token builtin class-name" style="color:rgb(255, 203, 107)">.</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_Ue-o">Copy</button></div></div><ul><li>Recursive copies on directories also work with <code>-r</code></li></ul><div class="codeBlockContainer_K1bP"><div class="codeBlockContent_hGly bash"><div tabindex="0" class="prism-code language-bash codeBlock_23N8 thin-scrollbar"><div class="codeBlockLines_39YC" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token function" style="color:rgb(130, 170, 255)">scp</span><span class="token plain"> –r hbar.ices.utexas.edu:public_html ~/tmp</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_Ue-o">Copy</button></div></div><h5><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="rsync---a-fast-versatile-remote-and-local-file-copying-tool"></a>rsync - a fast, versatile, remote (and local) file-copying tool<a class="hash-link" href="#rsync---a-fast-versatile-remote-and-local-file-copying-tool" title="Direct link to heading">#</a></h5><ul><li>Example 1. Synchronize Two Directories in a Local Server</li></ul><div class="codeBlockContainer_K1bP"><div class="codeBlockContent_hGly bash"><div tabindex="0" class="prism-code language-bash codeBlock_23N8 thin-scrollbar"><div class="codeBlockLines_39YC" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token function" style="color:rgb(130, 170, 255)">rsync</span><span class="token plain"> -zvr /path/to/source /path/to/copy</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_Ue-o">Copy</button></div></div><ul><li><p><code>z</code> - enable compression</p></li><li><p><code>v</code> - verbose</p></li><li><p><code>r</code> -recursive</p></li><li><p>Example 2. Preserve timestamps during Sync using rsync <code>–a</code></p></li></ul><div class="codeBlockContainer_K1bP"><div class="codeBlockContent_hGly bash"><div tabindex="0" class="prism-code language-bash codeBlock_23N8 thin-scrollbar"><div class="codeBlockLines_39YC" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token function" style="color:rgb(130, 170, 255)">rsync</span><span class="token plain"> -azv /path/to/source /path/to/copy</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_Ue-o">Copy</button></div></div><ul><li><code>-a</code> enters archive mode</li><li>Archive mode does the following:</li><li>Recursive mode</li><li>Preserves symbolic links</li><li>Preserves permissions</li><li>Preserves timestamp</li><li>Preserves owner and group</li></ul><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="back-to-the-tutorial"></a>Back to the Tutorial<a class="hash-link" href="#back-to-the-tutorial" title="Direct link to heading">#</a></h3><p>So we&#x27;ve got the data. But how do we get the software we need to do anything with it?</p><p>We&#x27;ll first use <a href="https://lmod.readthedocs.io/en/latest/" target="_blank" rel="noopener noreferrer">Lmod: A New Environment Module System</a></p><p>Some helpful commands:</p><div class="codeBlockContainer_K1bP"><div class="codeBlockContent_hGly bash"><div tabindex="0" class="prism-code language-bash codeBlock_23N8 thin-scrollbar"><div class="codeBlockLines_39YC" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">login $ module list </span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic"># What modules do I currently have enabled?</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">login $ module load subread </span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic"># Load the subread module</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">login $ module unload subread  </span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic"># Unload the subread module</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">login $ module unload anaconda3  </span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic"># Load the anaconda3 module</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">login $ module save default </span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic"># Save the current module stack to the default stack</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_Ue-o">Copy</button></div></div><p>So let&#x27;s actually run <code>featurecounts</code> and get the counts for our genes!</p><div class="codeBlockContainer_K1bP"><div class="codeBlockContent_hGly bash"><div tabindex="0" class="prism-code language-bash codeBlock_23N8 thin-scrollbar"><div class="codeBlockLines_39YC" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">login $ srun -p normal --time </span><span class="token number" style="color:rgb(247, 140, 108)">1</span><span class="token plain">-00 --mem</span><span class="token operator" style="color:rgb(137, 221, 255)">=</span><span class="token plain">8G --ntasks</span><span class="token operator" style="color:rgb(137, 221, 255)">=</span><span class="token number" style="color:rgb(247, 140, 108)">8</span><span class="token plain"> --pty </span><span class="token function" style="color:rgb(130, 170, 255)">bash</span><span class="token plain"> -i</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">compute $ ml load subread</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">compute $ featureCounts -t exon -g gene_id -a genome.gtf -o counts.txt test.paired_end.sorted.bam</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_Ue-o">Copy</button></div></div><p>Pop open <code>counts.txt</code> with <code>cat counts.txt</code> and check out the data!</p><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="sbatch"></a><code>sbatch</code><a class="hash-link" href="#sbatch" title="Direct link to heading">#</a></h3><p>So the interactive login for jobs is great, when no one else is using the cluster! Why should we have to wait around in line all day to get our time on a compute node? What if we could tell the cluster what we wanted to do and it would schedule the jobs for us?</p><blockquote><p>You use the sbatch command in front of the script you actually want to run. sbatch then puts your job into the job queue. The job scheduler looks at the current status of the whole system and will assign the first job in the queue to a node that is free in terms of computational load. If all machines are busy, yours will wait. But your job will sooner or later get assigned to a free node. <sup id="fnref-6"><a href="#fn-6" class="footnote-ref">6</a></sup></p></blockquote><h4><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="example-script"></a>Example script<a class="hash-link" href="#example-script" title="Direct link to heading">#</a></h4><div class="codeBlockContainer_K1bP"><div class="codeBlockContent_hGly bash"><div tabindex="0" class="prism-code language-bash codeBlock_23N8 thin-scrollbar"><div class="codeBlockLines_39YC" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token shebang important">#!/bin/bash</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic"># Set a name for the job (-J or --job-name).</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic">#SBATCH --job-name=tutorial</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic"># Set the file to write the stdout and stderr to (if -e is not set; -o or --output).</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic">#SBATCH --output=logs/%x-%j.log</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic"># Set the number of cores (-n or --ntasks).</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic">#SBATCH --ntasks=8</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic"># Force allocation of the two cores on ONE node.</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic">#SBATCH --nodes=1</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic"># Set the total memory. Units can be given in T|G|M|K.</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic">#SBATCH --mem=8G</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic"># Optionally, set the partition to be used (-p or --partition).</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic">#SBATCH --partition=medium</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic"># Set the expected running time of your job (-t or --time).</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic"># Formats are MM:SS, HH:MM:SS, Days-HH, Days-HH:MM, Days-HH:MM:SS</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic">#SBATCH --time=30:00</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token builtin class-name" style="color:rgb(255, 203, 107)">export</span><span class="token plain"> </span><span class="token assign-left variable" style="color:rgb(191, 199, 213)">TMPDIR</span><span class="token operator" style="color:rgb(137, 221, 255)">=</span><span class="token plain">/fast/users/</span><span class="token variable" style="color:rgb(191, 199, 213)">${</span><span class="token variable environment constant" style="color:rgb(130, 170, 255)">USER</span><span class="token variable" style="color:rgb(191, 199, 213)">}</span><span class="token plain">/scratch/tmp</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token function" style="color:rgb(130, 170, 255)">mkdir</span><span class="token plain"> -p </span><span class="token variable" style="color:rgb(191, 199, 213)">${TMPDIR}</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_Ue-o">Copy</button></div></div><div class="admonition admonition-info alert alert--info"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="14" height="16" viewBox="0 0 14 16"><path fill-rule="evenodd" d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg></span>info</h5></div><div class="admonition-content"><p>The lines starting with #SBATCH are actually setting parameters for a sbatch command, so #SBATCH --job-name=tutorial is equal to sbatch --job-name=tutorial.</p></div></div><h4><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="turning-our-featurecount-command-into-a-script"></a>Turning our featurecount command into a script<a class="hash-link" href="#turning-our-featurecount-command-into-a-script" title="Direct link to heading">#</a></h4><p>Open up a new file in your text editor</p><div class="codeBlockContainer_K1bP"><div style="color:#bfc7d5;background-color:#292d3e" class="codeBlockTitle_eoMF">submit_job.sh</div><div class="codeBlockContent_hGly bash"><div tabindex="0" class="prism-code language-bash codeBlock_23N8 thin-scrollbar codeBlockWithTitle_2JqI"><div class="codeBlockLines_39YC" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token shebang important">#!/bin/bash</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic">#SBATCH --job-name=featurecounts</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic">#SBATCH --ntasks=8</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic">#SBATCH --nodes=1</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic">#SBATCH --mem=8G</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic">#SBATCH --partition=normal</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic">#SBATCH --time=30:00</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">ml load subread</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">featureCounts -t exon -g gene_id -a genome.gtf -o counts.txt test.paired_end.sorted.bam</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_Ue-o">Copy</button></div></div><p>And then submit it to the cluster with</p><div class="codeBlockContainer_K1bP"><div class="codeBlockContent_hGly bash"><div tabindex="0" class="prism-code language-bash codeBlock_23N8 thin-scrollbar"><div class="codeBlockLines_39YC" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">login $ sbatch submit_job.sh</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">login $ </span><span class="token function" style="color:rgb(130, 170, 255)">cat</span><span class="token plain"> *.out</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_Ue-o">Copy</button></div></div><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="commit-your-changes"></a>Commit your Changes!<a class="hash-link" href="#commit-your-changes" title="Direct link to heading">#</a></h3><p>Make sure to gitignore your input data and any results you produced! Along with
any log files from sbatch.
It&#x27;s up to you whether you want to create a make rule for <code>sbatch</code> or just include it in the <code>README</code></p><h1><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2LWZ" id="further-reading"></a>Further Reading<a class="hash-link" href="#further-reading" title="Direct link to heading">#</a></h1><ul><li><a href="https://bihealth.github.io/bih-cluster/slurm/cheat-sheet/" target="_blank" rel="noopener noreferrer">Slurm Cheatsheet</a></li><li><a href="http://docs.oithpc.utdallas.edu/" target="_blank" rel="noopener noreferrer">UTDallas HPC Cluster Users Guide</a></li><li><a href="https://docs.ycrc.yale.edu/clusters-at-yale/" target="_blank" rel="noopener noreferrer">Yale HPC Docs</a></li></ul><div class="footnotes"><hr><ol><li id="fn-1"><a href="https://portal.tacc.utexas.edu/user-guides/lonestar5" target="_blank" rel="noopener noreferrer">TACC Lonestar5 User guide</a><a href="#fnref-1" class="footnote-backref">↩</a></li><li id="fn-2"><a href="https://docs.ycrc.yale.edu/clusters-at-yale/" target="_blank" rel="noopener noreferrer">Yale HPC Docs</a><a href="#fnref-2" class="footnote-backref">↩</a></li><li id="fn-3"><a href="http://docs.oithpc.utdallas.edu/" target="_blank" rel="noopener noreferrer">UTDallas HPC Cluster Users Guide</a><a href="#fnref-3" class="footnote-backref">↩</a></li><li id="fn-4"><a href="https://www.biostarhandbook.com/common-data-types.html" target="_blank" rel="noopener noreferrer">Biostars handbook Common data types</a><a href="#fnref-4" class="footnote-backref">↩</a></li><li id="fn-5">[HPC Class SSH and Slurm lecture]<a href="#fnref-5" class="footnote-backref">↩</a></li><li id="fn-6"><a href="https://bihealth.github.io/bih-cluster/first-steps/" target="_blank" rel="noopener noreferrer">BIH HPC Docs</a><a href="#fnref-6" class="footnote-backref">↩</a></li></ol></div></div></article><div class="margin-vert--xl"><div class="row"><div class="col"><a href="https://github.com/Functional-Genomics-Lab/Applied-Genomics/edit/main/versioned_docs/version-21u/week_03/hpc.md" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="1.2em" width="1.2em" preserveAspectRatio="xMidYMid meet" role="img" viewBox="0 0 40 40" class="iconEdit_2_ui" aria-label="Edit page"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div></div></div><div class="margin-vert--lg"><nav class="pagination-nav" aria-label="Docs pages navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/Applied-Genomics/docs/21u/week_03/conda"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">« Conda</div></a></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/Applied-Genomics/docs/21u/week_03/assignment_03"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Assignment 3 »</div></a></div></nav></div></div></div><div class="col col--3"><div class="tableOfContents_35-E thin-scrollbar"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#whats-a-cluster" class="table-of-contents__link">What&#39;s a cluster?</a></li><li><a href="#slurm" class="table-of-contents__link">Slurm</a></li><li><a href="#tutorial" class="table-of-contents__link">Tutorial</a><ul><li><a href="#instant-gratification" class="table-of-contents__link">Instant Gratification</a></li><li><a href="#scratch" class="table-of-contents__link"><code>scratch</code></a></li><li><a href="#lets-get-some-data" class="table-of-contents__link">Let&#39;s get some data</a></li><li><a href="#back-to-the-tutorial" class="table-of-contents__link">Back to the Tutorial</a></li><li><a href="#sbatch" class="table-of-contents__link"><code>sbatch</code></a></li><li><a href="#commit-your-changes" class="table-of-contents__link">Commit your Changes!</a></li></ul></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="row footer__links"><div class="col footer__col"><h4 class="footer__title">Course</h4><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/Applied-Genomics/docs/overview">Overview</a></li></ul></div></div><div class="footer__bottom text--center"></div></div></footer></div>
<script src="/Applied-Genomics/assets/js/runtime~main.70778c13.js"></script>
<script src="/Applied-Genomics/assets/js/main.c419694f.js"></script>
</body>
</html>